guide/src/faq.md

```diff
@@ -15,9 +15,21 @@ PyO3 provides a struct [`GILOnceCell`] which works equivalently to `OnceCell` bu
 
 [`GILOnceCell`]: {{#PYO3_DOCS_URL}}/pyo3/once_cell/struct.GILOnceCell.html
 
-## I can't run `cargo test`: I'm having linker issues like "Symbol not found" or "Undefined reference to _PyExc_SystemError"!
+## I can't run `cargo test`; or I can't build in a Cargo workspace: I'm having linker issues like "Symbol not found" or "Undefined reference to _PyExc_SystemError"!
 
-Currently, [#340](https://github.com/PyO3/pyo3/issues/340) causes `cargo test` to fail with linking errors when the `extension-module` feature is activated. For now you can work around this by making the `extension-module` feature optional and running the tests with `cargo test --no-default-features`:
+Currently, [#340](https://github.com/PyO3/pyo3/issues/340) causes `cargo test` to fail with linking errors when the `extension-module` feature is activated. Linking errors can also happen when building in a cargo workspace where a different crate also uses PyO3 (see [#2521](https://github.com/PyO3/pyo3/issues/2521)). For now, there are two ways we can work around these issues.
+
+1. Make the `extension-module` feature optional. Build with `maturin develop --features "extension-module"`
+
+```toml
+[dependencies.pyo3]
+{{#PYO3_CRATE_VERSION}}
+
+[features]
+extension-module = ["pyo3/extension-module"]
+```
+
+2. Make the `extension-module` feature optional and default. Run tests with `cargo test --no-default-features`:
```

Comment:
I think we can also mention `[tool.maturin]` section in `pyproject.toml`, see https://github.com/PyO3/pyo3/discussions/2271#discussioncomment-2497276

```toml
[tool.maturin]
features = ["extension-module"]
```

Reply:
I didn't include that because I couldn't get it to work for me. When I had errors in the format of `pyproject.toml`, it wouldn't compile, but once I fixed them, I still got the warning from maturin that it wasn't compiling with the `extension-module` feature. And actually I just grepped the pyo3 repo and I can't find documentation explaining `pyproject.toml` anywhere either.

If we do add this as a third workaround, I think we need to be able to link to some documentation that explains `pyproject.toml` so it is clear _why_ this works too?

Reply:
https://maturin.rs/metadata.html#add-maturin-build-options, you need maturin 0.13 for `features` or `cargo-extra-args` on maturin 0.12.

Reply:
Okay, I was confused because I still get "Warning: You're building a library without activating pyo3's `extension-module` feature." but actually it does properly work. It is just that the warning is emitted before the pyproject.toml is parsed. I'll add this as a third workaround.

Reply:
Hmm, that warning might be a bug in maturin, can you create a issue in https://github.com/PyO3/maturin/issues? Thanks!

Reply:
Done! https://github.com/PyO3/maturin/issues/1029

And how does the PR look now?